lms/djangoapps/experiments/views.py

```diff
@@ -84,3 +91,26 @@ class ExperimentKeyValueViewSet(viewsets.ModelViewSet):
     permission_classes = (IsStaffOrReadOnly,)
     queryset = ExperimentKeyValue.objects.all()
     serializer_class = serializers.ExperimentKeyValueSerializer
+
+
+class UserMetaDataView(APIView):
+    authentication_classes = (JwtAuthentication, ExperimentCrossDomainSessionAuth,)
+    permission_classes = (IsStaffOrReadOnlyForSelf,)
+
+    def get(self, request, course_id=None, username=None):
+        """ Return user-metadata for the given course and user """
+        try:
+            user = get_user_by_username_or_email(username)
+        except User.DoesNotExist:
+            message = "Provided user is not found"
```

Comment:
Should we be more vague in the error messaging so that we don't give away information about which usernames have accounts and which don't?

Reply:
After the intended purpose of Optimizely using this new view for the experiment, an alternate use case could be our internal troubleshooting, in which case knowing whether it was the course vs the user that got a 404 is useful. The user who reaches this code is someone logged in with an internal edX user account. 

I think the only case where we'd want to suppress this information would be: a bad actor makes a (non-staff) account, finds this endpoint, and tries to look up another user. This user never reaches the code above; has_permissions is false in the IsStaffOrReadOnlyForSelf permission, so they already have a 403 Forbidden response before anything else happens.